.teamcityTest/Gradle_Check_Tests/CIConfigIntegrationTests.kt

```diff
@@ -230,6 +231,23 @@ class CIConfigIntegrationTests {
         }
     }
 
+    @Test
+    fun testsAreCorrectlyConfiguredForAllSubProjects() {
+        CIBuildModel().subProjects.filter {
+            !listOf(
+                "soak", // soak test
+                "distributions", // build distributions
+                "docs", // sanity check
+                "architectureTest" // sanity check
+            ).contains(it.name)
+        }.forEach {
+            val dir = getSubProjectFolder(it)
+            assertEquals(it.unitTests, File(dir, "src/test").isDirectory, "${it.name}'s unitTests is wrong!")
```

Comment:
How should we handle this between `release` and `master`? As long as we only add tests, we are good, but when we remove tests we will lose some coverage on `release`.

Reply:
Ah right... I forgot... thanks Stefan. We did only check for missing declarations and not for declarations that are "too much" because of the branches situation (see how `allSubprojectsDefineTheirUnitTestPropertyCorrectly` only checks that direction). :(

Reply:
I think it is better to relax the check here then. We only miss something if it gets removed, but that should usually not happen (very often). And if it is the case it is better to run more because of the release/master situation Stefan describes.

Maybe we can keep the other check in a comment so that one can run it manually from time to time. Or have a separate test that is ignored (on CI).

Reply:
We did pay overhead because of such lack of checking before. I noticed this issue because I saw some empty builds in `AllVersionsCrossVersionTest` (like [this](https://builds.gradle.org/viewType.html?buildTypeId=Gradle_Check_AllVersionsCrossVersion_11_pluginUse&tab=buildTypeHistoryList&branch_Gradle_Check_AllVersionsCrossVersion_11=master)). These builds were scheduled because `pluginUse` subproject's configuration was wrong.

Regarding the branch situation, IIUC, you mean:

On `master`/`release` we have `A`/`B` subprojects. Then we delete `B` subproject on master, now we have to delete `B` from pipeline because of this  check - now `release` loses `B`. This can be solved by bucket split - if we detect such situation on `release`, we add the deleted subproject into one other bucket.